src/poetry_plugin_export/exporter.py

```diff
@@ -98,17 +98,10 @@ def _export_requirements_txt(self, cwd: Path, output: IO | str) -> None:
         ops = solver.solve().calculate_operations()
         packages = sorted((op.package for op in ops), key=lambda pkg: pkg.name)
 
-        # Get project dependencies.
-        if self._groups is not None:
```

Comment:
I do not think this change is correct. The intended behaviour should be if groups are specified, export them. And if no groups are specified, default to `poetry export --with=default` to emulate current behavriour. I tink the  `self._poetry.package.without_optional_dependency_groups()` should be dropped and we should use `list(self._groups or ["default"])` instead.

The `dev=True` is superfluous, as we should really remove that option in poetry as well.

Reply:
If I haven't overlooked something, removing these lines is fine because they are duplicates (see L81-L86). As for the default behavior, I'll check if I can apply your proposal in the lines that are still existing.

At the moment the `dev=True` is necessary because otherwise depdendencies not in the default group are always dropped and tests fail. (Nevertheless, I'll check again after applying the proposed changes.)

Reply:
I restored the default behavior of `Exporter.export()` and verified that `dev=True` is required to get correct results because dev defaults to `False`.

However, the default behavior of `Exporter.export()` is not as relevant as I thought first because `groups` are always set by the command.

Reply:
Groups LGTM now.

Yes, the `dev=True` is required. It is fairly useless with group, I wonder if it is easier to just not add the parameter here, and update the `use-export-plugin` branch upstream to switch locker to default to `dev=True` behaviour? This will save us a deprecation cycle. Since the 1.0.0 release of the plugin requires the upstream branch thanks to https://github.com/python-poetry/poetry-plugin-export/blob/main/pyproject.toml#L19.

Reply:
https://github.com/python-poetry/poetry/pull/5413/commits/642775bf9c7ad2ac74e9495fd8290525bb9060c9